clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.dto.AuditDTO;
+import com.datastrato.gravitino.dto.CatalogDTO;
+import com.datastrato.gravitino.dto.requests.SchemaCreateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdatesRequest;
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.SchemaResponse;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.NonEmptySchemaException;
+import com.datastrato.gravitino.exceptions.SchemaAlreadyExistsException;
+import com.datastrato.gravitino.rel.Schema;
+import com.datastrato.gravitino.rel.SchemaChange;
+import com.datastrato.gravitino.rel.SupportsSchemas;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * BaseSchemaCatalog is the base abstract class for all the catalog with schema. It provides the
+ * common methods for managing schemas in a catalog. With {@link BaseSchemaCatalog}, users can list,
+ * create, load, alter and drop a schema with specified identifier.
+ */
+public abstract class BaseSchemaCatalog extends CatalogDTO implements SupportsSchemas {
```

Comment:
Must this class be public? Is it default enough?

Reply:
Down